crates/vm/errors.rs

```diff
@@ -96,18 +95,6 @@ pub enum StateProofsError {
     StorageProofNotFound(RevmAddress, RevmU256),
 }
 
-impl From<RevmError<StoreError>> for EvmError {
-    fn from(value: RevmError<StoreError>) -> Self {
-        match value {
-            RevmError::Transaction(err) => EvmError::Transaction(err.to_string()),
-            RevmError::Header(err) => EvmError::Header(err.to_string()),
-            RevmError::Database(err) => EvmError::DB(err.to_string()),
-            RevmError::Custom(err) => EvmError::Custom(err),
-            RevmError::Precompile(err) => EvmError::Precompile(err),
-        }
-    }
-}
```

Comment:
Would it be helpful to make this implementation generic over `E: Display`?
Something like `impl<E: Display> From<RevmError<E>> for EvmError `
That way we could use a single mapping and keep the same behaviour